python-frontend/src/main/java/org/sonar/python/semantic/SymbolTableBuilder.java

```diff
@@ -753,6 +754,26 @@ public void visitFunctionDef(FunctionDef functionDef) {
       super.visitFunctionDef(functionDef);
     }
 
+    @Override
+    public void visitAnnotatedAssignment(AnnotatedAssignment annotatedAssignment) {
+      if (annotatedAssignment.variable().is(Kind.NAME)) {
+        Name variable = (Name) annotatedAssignment.variable();
+        TypeAnnotation annotation = annotatedAssignment.annotation();
+        Symbol symbol = variable.symbol();
+        Optional.ofNullable(symbol)
+          .ifPresent(s -> {
+            Expression expression = annotation.expression();
+            if (expression instanceof HasSymbol) {
+              Symbol typeSymbol = ((HasSymbol) expression).symbol();
+              if (typeSymbol != null) {
+                ((SymbolImpl) symbol).setInferredType(new DeclaredType(typeSymbol, Collections.emptyList()));
```

Comment:
Note: after reviewing this [PR](https://github.com/SonarSource/sonar-python/pull/1654), I realize this is probably not the right way to create the `DeclaredType` and that I should probably use `InferredTypes#fromTypeAnnotation` instead.